.github/workflows/ci.yml

```
@@ -17,14 +17,20 @@ jobs:
   # Only then, normal testing proceeds
   unit-tests:
     needs: qa
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

By default, it will stop the process as soon as one of the combinations fail. You need to tell it not to fail fast - you want to run the whole matrix no matter which one fails.

```suggestion
      matrix:
        fail-fast: false
        os: [ubuntu-latest, macos-latest, windows-latest]
```